django/db/backends/base/schema.py

```diff
@@ -942,6 +942,9 @@ def _constraint_names(self, model, column_names=None, unique=None,
         Returns all constraint names matching the columns and conditions
         """
         column_names = list(column_names) if column_names else None
+        if column_names and self.connection.features.uppercases_column_names:
+            # Oracle's get_constraints() returns all columns as lower case.
```

Comment:
You're right about the `LOWER()` query I didn't look at it and only based my comment on the `uppercases_column_names` feature name.

The Oracle backend has a lot of issues in regard to cases, the fact that it uppercases all columns but inspects them as lowercased is completely nut.

I think we should adjust `get_constraints()` to preserve the column cases and uppercase all provided `column_names` if `uppercases_column_names`.